### PR TITLE
Update samples to support sandboxes properly and use the latest runti…

### PR DESCRIPTION
**Pull request by ichasepucks**
…me image

### CODE REVIEW

File: recipes/pyspark/pysparkretailapp/helper.py [hunk 1]
```diff
@@ -34,26 +34,38 @@ def load_dataset(config_properties, spark, task_id):
...
-    pd = spark.read.format(PLATFORM_SDK_PQS_PACKAGE) \
-        .option(query_options.userToken(), user_token) \
-        .option(query_options.serviceToken(), service_token) \
-        .option(query_options.imsOrg(), org_id) \
-        .option(query_options.apiKey(), api_key) \
-        .option(query_options.mode(), PLATFORM_SDK_PQS_INTERACTIVE) \
-        .option(query_options.datasetId(), dataset_id) \
-        .option(query_options.sandboxName(), sandbox_name) \
-        .load()
+    if sandbox_name:
```
Comment (ichasepucks) [inline recipes/pyspark/pysparkretailapp/helper.py:49]: This is not DRY and I don't like it but I could not figure out how to create an object that I could then apply the sandbox to conditionally and then execute `.load()`.  It gets a py4j binding exception.  I punted on it.

File: recipes/scala/src/main/scala/com/adobe/platform/ml/helper/Helper.scala [hunk 2]
```diff
@@ -54,18 +54,39 @@ class Helper {
...
+        .option(QSOption.userToken, userToken)
+        .option(QSOption.serviceToken, serviceToken)
+        .option(QSOption.imsOrg, orgId)
+        .option(QSOption.apiKey, apiKey)
+        // Batch mode can handle much larger datasets but but has a few minute
+        // additional startup time required.  Interactive can be faster to read
+        // smaller amounts of data but can be subject to timeouts if the amount
+        // of data is too large.
+        // See https://docs.adobe.com/content/help/en/experience-platform/data-science-workspace/authoring/spark.html#reading-a-dataset
+        .option(QSOption.mode, PLATFORM_SDK_PQS_BATCH)
```
Comment (ichasepucks) [inline recipes/scala/src/main/scala/com/adobe/platform/ml/helper/Helper.scala]: This also covers switching to batch mode so we ensure we have coverage of both interactive via pyspark and batch via scala.

File: recipes/scala/src/main/scala/com/adobe/platform/ml/helper/Helper.scala [hunk 3]
```diff
@@ -54,18 +54,39 @@ class Helper {
...
+      sparkSession.read.format(PLATFORM_SDK_PQS_PACKAGE)
+        .option(QSOption.userToken, userToken)
+        .option(QSOption.serviceToken, serviceToken)
+        .option(QSOption.imsOrg, orgId)
+        .option(QSOption.apiKey, apiKey)
+        .option(QSOption.mode, PLATFORM_SDK_PQS_BATCH)
+        .option(QSOption.datasetId, dataSetId)
+        .option(QSOption.sandboxName, sandboxName)
+        .load()
+    }
```
Comment (kratika2210) [inline recipes/scala/src/main/scala/com/adobe/platform/ml/helper/Helper.scala:76]: For DRY-ing this code- you can just use the `Reader` first and then `load()` later:
```
var dataFrameReader = sparkSession.read.format(PLATFORM_SDK_PQS_PACKAGE)
  .option(QSOption.userToken, userToken)
  .option(QSOption.serviceToken, serviceToken)
  .option(QSOption.imsOrg, orgId)
  .option(QSOption.apiKey, apiKey)
  .option(QSOption.mode, PLATFORM_SDK_PQS_BATCH)
  .option(QSOption.datasetId, dataSetId)

if(!sandboxName.isEmpty()) {
  dataFrameReader.option(QSOption.sandboxName, sandboxName)
}

var df = dataFrameReader.load()
  
```

File: recipes/pyspark/pysparkretailapp/helper.py [hunk 4]
```diff
@@ -34,26 +34,38 @@ def load_dataset(config_properties, spark, task_id):
...
+    else:
+        pd = spark.read.format(PLATFORM_SDK_PQS_PACKAGE) \
+            .option(query_options.userToken(), user_token) \
+            .option(query_options.serviceToken(), service_token) \
+            .option(query_options.imsOrg(), org_id) \
+            .option(query_options.apiKey(), api_key) \
+            .option(query_options.mode(), PLATFORM_SDK_PQS_INTERACTIVE) \
+            .option(query_options.datasetId(), dataset_id) \
+            .load()
+
```
Comment (kratika2210) [inline recipes/pyspark/pysparkretailapp/helper.py:68]: same comment here on DRY-ing

**Review by kratika2210 — CHANGES_REQUESTED** [review]
Made a suggestion for DRY-ing the code

**Review by jonaphin — APPROVED** [review]
conditionally approving based on @kratika2210 's suggestion

File: recipes/scala/src/main/scala/com/adobe/platform/ml/helper/Helper.scala (hunk 3 above)
Comment (ichasepucks) [reply recipes/scala/src/main/scala/com/adobe/platform/ml/helper/Helper.scala:76]: incorporated this for Scala.

File: recipes/pyspark/pysparkretailapp/helper.py (hunk 4 above)
Comment (ichasepucks) [reply recipes/pyspark/pysparkretailapp/helper.py:68]: Yeah I can't dry this up using the same technique before due to py4j failure.

**Review by kratika2210 — APPROVED** [review]

**Pull request merged**